openmc/mgxs/mgxs.py

```diff
@@ -0,0 +1,5909 @@
+                                         subdomains=[subdomain],
+                                         xs_type=xs_type, value='mean')
+                rel_err_xs = self.get_xs(nuclides=[nuclide],
+                                         subdomains=[subdomain],
+                                         xs_type=xs_type, value='rel_err')
+                rel_err_xs = rel_err_xs * 100.
+
+                if self.num_polar > 1 or self.num_azimuthal > 1:
+                    # Loop over polar, azimuthal, and energy group ranges
+                    for pol in range(len(pol_bins) - 1):
+                        pol_low, pol_high = pol_bins[pol: pol + 2]
+                        for azi in range(len(azi_bins) - 1):
+                            azi_low, azi_high = azi_bins[azi: azi + 2]
+                            string += '\t\tPolar Angle: [{0:5f} - {1:5f}]'.format(
+                                pol_low, pol_high) + \
+                                '\tAzimuthal Angle: [{0:5f} - {1:5f}]'.format(
+                                azi_low, azi_high) + '\n'
+                            for group in range(1, self.num_groups + 1):
+                                bounds = \
+                                    self.energy_groups.get_group_bounds(group)
+                                string += '\t' + template.format('', group,
+                                                                 bounds[0],
+                                                                 bounds[1])
+
+                                string += '{0:.2e} +/- {1:.2e}%'.format(
+                                    average_xs[pol, azi, group - 1],
+                                    rel_err_xs[pol, azi, group - 1])
+                                string += '\n'
+                            string += '\n'
+                else:
+                    # Loop over energy groups
+                    for group in range(1, self.num_groups + 1):
+                        bounds = self.energy_groups.get_group_bounds(group)
+                        string += template.format('', group, bounds[0],
+                                                  bounds[1])
+                        string += '{0:.2e} +/- {1:.2e}%'.format(
+                            average_xs[group - 1], rel_err_xs[group - 1])
+                        string += '\n'
+                string += '\n'
+            string += '\n'
+
+        print(string)
+
+    def build_hdf5_store(self, filename='mgxs.h5', directory='mgxs',
+                         subdomains='all', nuclides='all',
+                         xs_type='macro', row_column='inout', append=True,
+                         libver='earliest'):
+        """Export the multi-group cross section data to an HDF5 binary file.
+
+        This method constructs an HDF5 file which stores the multi-group
+        cross section data. The data is stored in a hierarchy of HDF5 groups
+        from the domain type, domain id, subd
```

Comment:
This same conditional is copied three times. Maybe this can be factored out into it's own method?